skops/io/_numpy.py

```diff
@@ -15,47 +16,53 @@ def ndarray_get_state(obj, dst):
         "__module__": get_module(type(obj)),
     }
 
-    try:
+    # First, try to save object with np.save and allow_pickle=False, which
+    # should generally work as long as the dtype is not object.
+    with suppress(ValueError):
```

Comment:
the suppress/return pattern makes this harder to understand. Any reason you don't want to stay with try/except? try/except would seem much more readable to me.

Reply:
I prefer early return to lengthy try...except blocks but I will change it back if you prefer the other way round.

Reply:
I agree with early returns usually, but this suppress/return is somewhat convoluted!